From 9cdd67ed25921d59ec11bc3c6fc30bd33b8559cf Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Wed, 06 Jan 2021 17:50:45 +0200
Subject: [PATCH] [SW-27979] i2c indirect: fix status alignment

When setting status RSDL there is an alignment issue. Fixed by aligning
status to 4 bytes.

Change-Id: I157c0ec97c88ceaa2377759ee204d4ef18e15c22
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/preboot/i2cslave_indirect.c b/preboot/i2cslave_indirect.c
index c12e6f1..c1b8fd8 100644
--- a/preboot/i2cslave_indirect.c
+++ b/preboot/i2cslave_indirect.c
@@ -39,7 +39,7 @@
 
 struct i2cslave_indirect_req {
 	struct hl_i2ci_req_cfg cfg;
-	struct hl_i2ci_req_sts sts;
+	struct hl_i2ci_req_sts sts __aligned(4);
 	struct i2csi_wdata_chunk_context chunk;
 	struct i2csi_wdata_read_context rd;
 	struct i2csi_wdata_write_context wd;
